File: src/parser/ruby/defvars.rb
#!/usr/bin/env ruby

# [https://stackoverflow.com/a/171011]
# [https://stackoverflow.com/a/55169109]
def platform()
	if RUBY_PLATFORM =~ /cygwin|mswin|mingw|bccwin|wince|emx/
		return "windows"
	end

	if RUBY_PLATFORM =~ /darwin/
		return "darwin"
	end

	return "linux"
end

# Builtin variables.
def builtins(cmdname)
	return {
		"HOME": Dir.home,
		"OS": platform(),
		"COMMAND": cmdname,
		"PATH": "~/.nodecliac/registry/" + cmdname
	}
end

File: src/parser/js/defvars.js
#!/usr/bin/env node

"use strict";

function platform() {
	let platform = process.platform;
	if (platform === "darwin") platform = "macosx";
	if (platform === "win32") platform = "windows";
	return platform;
}

// Builtin variables.
function builtins(cmdname) {
	return {
		HOME: require("os").homedir(),
		OS: platform(),
		COMMAND: cmdname,
		PATH: `~/.nodecliac/registry/${cmdname}`
	};
}

module.exports = { builtins };
